### DISCUSSION TITLE
services.netbox-housekeeping.depends_on contains an invalid type, it should be an array?

### DISCUSSION THREAD

**Discussion by cplex** [netbox-community/netbox-docker#923] [Getting Help]
When running 
docker-compose pull
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.netbox-housekeeping.depends_on contains an invalid type, it should be an array
services.netbox-worker.depends_on contains an invalid type, it should be an array

**Comment by tobiasge** [comment]
Please update your docker-compose version.

**Comment by YooBZH** [comment]
Hi,

FYI I got the same error and I am running the latest docker engine and docker-compose version.

`A fix for me, if you can call it this way, was to use the previous version of netbox-docker : 2.3.0`

I am no docker expert and I do not have time to investigate it today but I will try to have a look at it in the coming days (but to be honest I think I already did everything I could think about).

what I did is follow the very basic installation steps from the readme.md of the 'release' branch.

docker --version → Docker version 20.10.5+dfsg1, build 55c4c88
docker-compose --version → docker-compose version 1.25.0, build unknown
docker-compose version : 
```
docker-compose version 1.25.0, build unknown
docker-py version: 4.1.0
CPython version: 3.9.2
OpenSSL version: OpenSSL 1.1.1n  15 Mar 2022
```

docker-compose --verbose pull :
```
compose.config.config.find: Using configuration files: ./docker-compose.yml,./docker-compose.override.yml
ERROR: compose.cli.main.main: The Compose file './docker-compose.yml' is invalid because:
services.netbox-housekeeping.depends_on contains an invalid type, it should be an array
services.netbox-worker.depends_on contains an invalid type, it should be an array
```

Same goes after an 'apt update && apt upgrade'
Same when trying to upgrade docker-compose specifically, I got : docker-compose is already the newest version (1.25.0-1).

Running on up-to-date Debian 11.6

I tried the same as the root user just to be sure : same output.
I also tried to change the "version" at the top of docker-compose.yml and docker-compose.override.yml file to everything from 3.1 to 3.9.

**Reply by tobiasge** [reply]
`1.25.0` is NOT the current version of docker-compose. See here: https://github.com/docker/compose/releases

**Reply by YooBZH** [reply]
Damn Debian I guess, thanks for the answer @tobiasge, i'll use 2.3.0 for now.

**Comment by tomehb** [comment]
On Debian 11, you could always...

   -  apt-get install docker-compose=1.27.4-3~bpo11+1


On Tue, 10 Jan 2023 at 16:32, YooBZH ***@***.***> wrote:

> Damn Debian I guess, thanks for the answer @tobiasge
> <https://github.com/tobiasge>, i'll stuck with 2.3.0 for now.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/netbox-community/netbox-docker/discussions/923#discussioncomment-4647362>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKXDFPJIPEZ3HVBCBCJOTDWRWFLTANCNFSM6AAAAAATUTNBGY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <netbox-community/netbox-docker/repo-discussions/923/comments/4647362@
> github.com>
>


-- 
---------------------------------------
Thomas

**Reply by Naaktgeboren** [reply]
> When running docker-compose pull ERROR: The Compose file './docker-compose.yml' is invalid because: services.netbox-housekeeping.depends_on contains an invalid type, it should be an array services.netbox-worker.depends_on contains an invalid type, it should be an array

I encountered exactly the same error today on my Ubuntu20.04 container (yes, I am nesting a docker container inside my proxmox CT). And it seems that apt update and apt upgrade can only bring docker-compose to version 1.25.0, but not anything higher, whilst Netbox docs clearly says a minimum of 1.28.0 is required. Hence, I used methods in this linked [article ](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-compose-on-ubuntu-20-04)to upgrade docker-compose to version 1.29.0.

And then the docker-compose pull doesn't generate the error shown in the title again. Hopefully this can help.

(But I encountered another set of errors which deserve their own thread for discussion).
